### PR TITLE
Proper state management for AES GCM and CCM

### PR DESCRIPTION
**Pull request by mamckee**
The SymCrypt unit tests with OpenSSL and SCOSSL revealed an edge case when calling `EVP_CipherInit_ex2` and passing the IV length as a parameter alongside the IV. The SCOSSL implementation unconditionally freed the set IV if an IV length was set, even if the IV length did not change. If the IV length matches what's set, then the IV should not be reset.

This PR also adds state checks for the IV and tag in CCM to bring the behavior and failure points more in line with the default provider.

### CODE REVIEW

File: SymCryptProvider/src/keymgmt/p_scossl_rsa_keymgmt.c [hunk 1]
```diff
@@ -1271,6 +1271,7 @@ static SCOSSL_STATUS p_scossl_rsa_keymgmt_import(_Inout_ SCOSSL_PROV_RSA_KEY_CTX
     ret = SCOSSL_SUCCESS;
 
 cleanup:
+    OPENSSL_free(pbPrivateExponent);
```
Comment (samuel-lee-msft) [inline SymCryptProvider/src/keymgmt/p_scossl_rsa_keymgmt.c:1274]: >OPENSSL_free [](http://example.com/codeflow?start=4&length=12)

Should this have been `OPENSSL_secure_malloc`ed and be `OPENSSL_secure_clear_free`ed?

File: SymCryptProvider/src/keymgmt/p_scossl_rsa_keymgmt.c [hunk 2]
```diff
@@ -1271,6 +1271,7 @@ static SCOSSL_STATUS p_scossl_rsa_keymgmt_import(_Inout_ SCOSSL_PROV_RSA_KEY_CTX
...
     OPENSSL_free(pbModulus);
     OPENSSL_free(ppbPrimes[0]);
     OPENSSL_free(ppbPrimes[1]);
```
Comment (samuel-lee-msft) [inline SymCryptProvider/src/keymgmt/p_scossl_rsa_keymgmt.c:1277]: Ditto for the private primes.

**Review by samuel-lee-msft — APPROVED** [review]
LGTM modulo comments

File: SymCryptProvider/src/keymgmt/p_scossl_rsa_keymgmt.c (hunk 1 above)
Comment (mamckee) [reply SymCryptProvider/src/keymgmt/p_scossl_rsa_keymgmt.c:1274]: Yeah, same for the primes.

**Review by samuel-lee-msft — APPROVED** [review]
